src/main/java/org/spongepowered/common/mixin/core/server/MixinPlayerList.java

```diff
@@ -529,9 +529,31 @@ public EntityPlayerMP recreatePlayerEntity(EntityPlayerMP playerIn, int targetDi
         // Sponge - Vanilla does this before recreating the player entity. However, we need to determine the bed location
         // before respawning the player, so we know what dimension to spawn them into. This means that the bed location must be copied
         // over to the new player
+        // [Extended, Revised, 28.01.2018, MC 1.12.2, API 7.1.0] Fix for: https://github.com/SpongePowered/SpongeCommon/issues/1719
+        // First revision: Xakep_SDK, 28.01.18. MC Version: 1.12.2, API version: 7.1.0
+        // START REVISION
+        // This gets target world bed spawn location
+        // and saves original player dimension for RespawnPlayerEvent event.
+        // We're getting target world because of:
+        // 1. When we're calling EntityUtil#getPlayerRespawnLocation(EntityPlayerMP playerIn, @Nullable WorldServer targetWorld);
+        // at the end of the method we have if(bedPos != null) ...
+        // blockPos - actual bed location for target world, that we got here.
+        // At the end of the if we can see:
+        // 1. Change player dim to target world dim
+        // 2. Change player bedPos to actual one (i don't know, why this happens, maybe a bigger kludge-system)
+        // 3. Reset player dim to old one
+        // This changes bedPos for TARGET world
+        // My changes: i'm setting players dim to target one,
+        // then we can get right bed position(if present)
+        // and set it to new players bed position
+        int tmp = playerIn.dimension;
+        playerIn.dimension = targetDimension;
         if (playerIn.getBedLocation() != null) {
             newPlayer.setSpawnPoint(playerIn.getBedLocation(), playerIn.isSpawnForced());
         }
+        // here i'm returning back player dim for RespawnPlayerEvent
```

Comment:
Eliminate ownership, just say `// Set player dim back from the RespawnPlayerEvent`

Reply:
Fixed in next commit